test/integration/pack/test_pack_docker.py

```diff
@@ -203,3 +207,86 @@ def test_image_tag_without_git(cartridge_cmd, project_without_dependencies, tmpd
     rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
     assert rc == 0
     assert 'Created result image with tags {}'.format(expected_image_tags) in output
+
+
+def test_customized_environment_variables(cartridge_cmd, project_without_dependencies, docker_client, request, tmpdir):
+    project = project_without_dependencies
+    replace_project_file(project, 'init.lua', INIT_PRINT_ENV_FILEPATH)
+    os.remove(os.path.join(project.path, 'Dockerfile.cartridge'))
+
+    cmd = [
+        cartridge_cmd,
+        "pack", "docker",
+        "--tag", project.name,
+        project.path,
+    ]
+
+    process = subprocess.run(cmd, cwd=tmpdir)
+    assert process.returncode == 0
+
+    instance_name = 'instance-1'
+    http_port = '8182'
+    advertise_port = '3302'
+
+    # Custom TARANTOOL_WORKDIR, TARANTOOL_PID_FILE and TARANTOOL_CONSOLE_SOCK
+    workdir = "test_workdir"
+    pidfile = "test_pidfile"
+    console_sock = "test_console_sock"
+
+    environment = [
+        f"TARANTOOL_APP_NAME={project.name}",
+        f"TARANTOOL_INSTANCE_NAME={instance_name}",
+        f"TARANTOOL_ADVERTISE_URI={advertise_port}",
+        f"TARANTOOL_HTTP_PORT={http_port}",
+        f"TARANTOOL_WORKDIR={workdir}",
+        f"TARANTOOL_PID_FILE={pidfile}",
+        f"TARANTOOL_CONSOLE_SOCK={console_sock}",
+    ]
+
+    container = docker_client.containers.run(
+        project.name,
+        environment=environment,
+        ports={http_port: http_port},
+        name='{}-{}'.format(project.name, instance_name),
+        detach=True,
+    )
+
+    request.addfinalizer(lambda: container.remove(force=True))
+
+    assert container.status == 'created'
+    container_message = f"{console_sock}\n{workdir}\n{pidfile}\n"
+
+    wait_for_container_start(container, time.time(), message=container_message)
+    container.stop()
+
+    # Custom CARTRIDGE_RUN_DIR and CARTRIDGE_DATA_DIR
```

Comment:
In fact, this is other test. It would be great to make it separated function.

Both tests use common image for tests that is in fact `docker_image` fixture. You can simply re-use it in two tests (look at the tests above).